homeassistant/core.py

```diff
@@ -1807,6 +1807,8 @@ def __init__(self, hass: HomeAssistant) -> None:
         self.internal_url: str | None = None
         self.external_url: str | None = None
         self.currency: str = "EUR"
+        self.country: str | None = None
+        self.language: str = "en"
```

Comment:
Should this not default to the language returned by the system?

```
>>> import locale
>>> locale.getdefaultlocale()
('en_US', 'UTF-8')
>>> lang = locale[0].partition("_")[0]
>>> lang = "en" if not lang or lang == "C" else lang
```

Reply:
I don't think the default locale of the machine on which Home Assistant is running on can be expected to be relevant or a good default, this should be set during onboarding instead based on the geographical location of the installation which the user configures and which defaults to location derived from the external IP address.

Reply:
Yes, it will probably complicate tests (you might have to add two tests with a mock for locale.getdefaultlocale(), or move the indirection where the default is determined outside of import scope), and it's also possible that the user's system's default language is not one of the valid translation choices.

That's part of making it work, no?

If I set up my system to work in French, and all my apps start in French, why should Home Assistant start up in English and then demand I manually set it to French?

Reply:
@emontnemery If the system should have its own default locale set up — for example, HASS OS — then the system should do that as part of the first startup scripts routine.  Then that information will be likely correct upon first HA start.  There are a number of reasons why this is a better choice than showing the UI in English or attempting autodetection as HA first starts up.

Of course, the information is likely to be correct already for people using HA Core, since they set up their systems by hand.

Reply:
The by far most popular installation types are HASS OS and container, none of those will set the OS locale according to the user's location as I understand it, meaning it's really not helpful to set the default country and language according to OS settings.

This is similar to how Home Assistant does not care about the time zone set in the OS, it has its own setting for that.

Reply:
As far as I know, _even in HASS OS_, the default time zone is to use the system time zone ("don't override the system").

Why can't that be the default also for country and language?

Reply:
We should not default to the system setting. We don't do that for other settings either and we should stay consistent. If we want to do that, we could explore it in another PR but I think the juice is not worth the squeeze as only a tiny minority is using a direct core installation.